src/Microsoft.Azure.WebJobs.Logging/Internal/LogReader.cs

```diff
@@ -155,20 +184,186 @@ public LogReader(CloudTable table)
         {
             TableQuery<RecentPerFuncEntity> rangeQuery = RecentPerFuncEntity.GetRecentFunctionsQuery(queryParams);
 
-            CancellationToken cancellationToken;
-            TableContinuationToken realContinuationToken = Utility.DeserializeToken(continuationToken);
-            var segment = await _instanceTable.SafeExecuteQuerySegmentedAsync<RecentPerFuncEntity>(
-                rangeQuery, 
-                realContinuationToken, 
-                cancellationToken);
+            var iter = new EpochTableIterator(_tableLookup);
+            var results = await iter.SafeExecuteQuerySegmentedAsync<RecentPerFuncEntity>(rangeQuery, queryParams.Start, queryParams.End);
+
+            return results.As<IRecentFunctionEntry>();
+        }
+
+        // Helper to run queries which can span multiple tables.         
+        private class EpochTableIterator
```

Comment:
Fair - it's not IEnumerable. But it does take in a continuation token and support iteration via continuation tokens. . 
it's internal, so we can change later. 

---
In reply to: [82881404](https://github.com/Azure/azure-webjobs-sdk/pull/862#discussion_r82881404) [](ancestors = 82881404)